File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/flyweight/lab/SingleJourney.java
// SingleJourney.java

import java.io.*;

public class SingleJourney extends Journey
{
	public SingleJourney(String aAddress)
	{
		if (aAddress == null || aAddress.equals(""))
		{
			throw new DpException("SingleJourney.SingleJourney(String aAddress) -- aAddress == null || aAddress.equals(\"\").");
		}
		myAddress = aAddress;
	}

	public void output(PrintWriter aPrintWriter)
	{
		if (aPrintWriter == null)
		{
			throw new DpException("void SingleJourney.output(PrintWriter aPrintWriter) -- aPrintWriter == null.");
		}

		aPrintWriter.println("single journey to " + myAddress + ".");
	}

	private String myAddress;
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/flyweight/lab/single_journey.cpp
// single_journey.cpp
#include "dp_exception.h"
#include "single_journey.h"
single_journey::single_journey(const std::string & a_address)
{
	if (a_address == "")
	{
		throw dp_exception("single_journey::single_journey(const std::string & a_address) -- a_address == \"\".");
	}
	my_address = a_address;
}

void single_journey::output(std::ostream & a_ostream) const
{
	a_ostream << "single journey to " << my_address << ".\n";
}
